### PR TITLE
Append request_id header

### PR DESCRIPTION
**Pull request by adorechic**
Set X-Request-Id header if there is an id on thread local.

### CODE REVIEW

**Comment by sorah** [comment]
I guess there's better name than `pass_over`...

- propagate
- pass
- ...

File: spec/garage_client/request/pass_over_request_id_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require "spec_helper"
+
+describe GarageClient::Request::PassOverRequestId do
+  let(:client) do
+    GarageClient::Client.new
+  end
+
+  before do
+    Thread.current[:request_id] = 'request_id'
```
Comment (eagletmt) [inline spec/garage_client/request/pass_over_request_id_spec.rb]: `Thread.current[:request_id]` should be restored

Comment (adorechic) [inline spec/garage_client/request/pass_over_request_id_spec.rb]: you mean `Thread.current[:request_id]` should be restored to be an original value?

```ruby
around do |example|
  original = Thread.current[:request_id]
  Thread.current[:request_id] = 'request_id'
  example.run
  Thread.current[:request_id] =  original
end
```

Comment (eagletmt) [inline spec/garage_client/request/pass_over_request_id_spec.rb]: yes

**Comment by adorechic** [comment]
Renamed "pass over" to "propagate"
6a921a634927661184e991ffc810a254b02a0eb1

File: spec/garage_client/request/pass_over_request_id_spec.rb (hunk 1 above)
Comment (adorechic) [inline spec/garage_client/request/pass_over_request_id_spec.rb]: 187d415235e6d09733f2258b1a714109fb398e51

File: spec/garage_client/request/propagate_request_id_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  it 'sends request_id via header' do
+    stub_get("/examples").with(headers: { 'HTTP_X_REQUEST_ID' => 'request_id'})
+    expect { client.get("/examples") }.not_to raise_error
+  end
+
+  context 'without request_id' do
+    before do
+      Thread.current[:request_id] = nil
```
Comment (taiki45) [inline spec/garage_client/request/propagate_request_id_spec.rb]: I think 187d415 should cover this case too.

File: spec/garage_client/request/propagate_request_id_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  around do |example|
+    original = Thread.current[:request_id]
+    Thread.current[:request_id] = 'request_id'
+    example.run
+    Thread.current[:request_id] = original
+  end
+
+  it 'sends request_id via header' do
+    stub_get("/examples").with(headers: { 'HTTP_X_REQUEST_ID' => 'request_id'})
```
Comment (taiki45) [inline spec/garage_client/request/propagate_request_id_spec.rb]: [nits] missing a space near braces.

File: spec/garage_client/request/propagate_request_id_spec.rb (hunk 2 above)
Comment (adorechic) [inline spec/garage_client/request/propagate_request_id_spec.rb]: Top level around block seems to run after this nested context.

File: spec/garage_client/request/propagate_request_id_spec.rb (hunk 3 above)
Comment (adorechic) [inline spec/garage_client/request/propagate_request_id_spec.rb]: fixed 51ef40e1dfaa0d2273b5e41686f3d9dd9ea3d384

Comment (adorechic) [inline spec/garage_client/request/propagate_request_id_spec.rb]: oops, f6b54c67141a161a65bc994dbd56e014fc4310c2

**Comment by taiki45** [comment]
This change looks good to me.

**Pull request merged**